src/torcontrol.cpp

```diff
@@ -524,10 +573,7 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
         // Now that we know Tor is running setup the proxy for onion addresses
         // if -onion isn't set to something else.
         if (gArgs.GetArg("-onion", "") == "") {
-            CService resolved(LookupNumeric("127.0.0.1", 9050));
-            proxyType addrOnion = proxyType(resolved, true);
-            SetProxy(NET_ONION, addrOnion);
-            SetReachable(NET_ONION, true);
+            _conn.Command("GETINFO net/listeners/socks", std::bind(&TorController::get_socks_cb, this, std::placeholders::_1, std::placeholders::_2));
```

Comment:
Minor Nit  it does not account for early user -proxy=  settings and would overrule them.

Please consider the litle patch ff 
That would also fix finally  #14722 and make #19358  obsolete.

``` Diff
diff --git a/src/torcontrol.cpp b/src/torcontrol.cpp
index 2871f9030..dff873ce9 100644
--- a/src/torcontrol.cpp
+++ b/src/torcontrol.cpp
@@ -573,7 +573,8 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
 
         // Now that we know Tor is running setup the proxy for onion addresses
         // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        proxyType addrOnion;
+        if (gArgs.GetArg("-onion", "") == "" && !GetProxy(NET_ONION, addrOnion)) {
             _conn.Command("GETINFO net/listeners/socks", std::bind(&TorController::get_socks_cb, this, std::placeholders::_1, std::placeholders::_2));
         }

Reply:
It would be incorrect behaviour.